lib/mods_display/fields/imprint.rb

```diff
@@ -42,167 +42,175 @@ def dates(element)
-    end
 
-    def process_encoded_dates(date_fields)
-      date_fields.map do |date_field|
-        if date_is_w3cdtf?(date_field)
-          process_w3cdtf_date(date_field)
-        elsif date_is_iso8601?(date_field)
-          process_iso8601_date(date_field)
-        else
-          date_field
-        end
+      # Value of element's "point" attribute; nil if empty or not present.
+      # Points are "start" or "end" (if part of a date range)
+      def point
+        @element.attributes['point']&.value.presence
       end
-    end
 
-    def join_date_ranges(date_fields)
-      if dates_are_range?(date_fields)
-        start_date = date_fields.find { |d| d.attributes['point'] && d.attributes['point'].value == 'start' }
-        end_date = date_fields.find { |d| d.attributes['point'] && d.attributes['point'].value == 'end' }
-        date_fields.map do |date|
-          date = date.clone # clone the date object so we don't append the same one
-          if normalize_date(date.text) == normalize_date(start_date.text)
-            date.content = [start_date.text, end_date.text].join('-')
-            date
-          elsif normalize_date(date.text) != normalize_date(end_date.text)
+      # Value of element's "qualifier" attribute; nil if empty or not present.
+      # Qualifiers are "approximate", "questionable", or "inferred"
+      def qualifier
+        @element.attributes['qualifier']&.value&.downcase.presence
+      end
+
+      # Value of element's "encoding" attribute; nil if empty or not present.
+      # Encodings are "w3cdtf", "iso8601", or "edtf"
+      def encoding
+        @element.attributes['encoding']&.value&.downcase.presence
+      end
+
+      # Element text reduced to digits and hyphen. Captures date ranges and
+      # negative (B.C.) dates. Used for comparison/deduping.
+      def base_value
+        @element.text.scan(/[\d-]/).join
+      end
+
+      # Human-formatted version of the date.
+      def decoded_value
+        date = @element.text.strip
+
+        decoded_date = if encoding == 'w3cdtf'
+          begin
+            if date =~ /^\d{4}-\d{2}-\d{2}$/
+              Date.parse(date).strftime('%B %e, %Y')
+            elsif date =~ /^\d{4}-\d{2}$/
+              Date.parse("#{date}-01").strftime('%B %Y')
+            else
+              date
+            end
+          rescue
             date
           end
-        end.compact
-      elsif dates_are_open_range?(date_fields)
-        start_date = date_fields.find { |d| d.attributes['point'] && d.attributes['point'].value == 'start' }
-        date_fields.map do |date|
-          date = date.clone # clone the date object so we don't append the same one
-          date.content = "#{start_date.text}-" if date.text == start_date.text
+            elsif encoding == 'iso8601'
+              begin
+                Date.iso8601(date).strftime('%B %e, %Y')
+              rescue
+                date
+              end
+            else
+              date
+          end
```

Comment:
I'd be tempted to decompose this section to something like the following, adding private methods to handle the specifics of decoding `w3cdtf` and `iso8601` types:

```
        decoded_date = if encoding == 'w3cdtf'
                         decode_w3cdtf(date)
                       elsif encoding == 'iso8601'
                         decode_iso8601(date)
                       else
                         date
                       end
```

Reply:
makes sense, since I think #10 might mean we want a `decode_edtf` too.

Reply:
@corylown made some changes, lmk what you think

Reply:
👍 -- much easier to understand and good use of `case`